### PR TITLE
Yet another infrastructure update

### PR DESCRIPTION
**Pull request by OoLunar**
Update csproj settings
- Generate reference assemblies once more
- Begone deprecated PackageIconUrl
- Remove default behavior
- Remove complicated targets to disable the js import
- Sort all XML keys alphabetically
- Make all csproj not packable by default
- Use project root due to MSBuild's non-existant concept of relative files
- Update MSE.Logging.Abstractions from 7.0.0 to 7.0.1
- Remove unused DotSettings file

### CODE REVIEW

**Review by InFTord — APPROVED** [review]
LGTM

File: Directory.Build.props [hunk 1]
```diff
@@ -1,73 +1,36 @@
...
+    <IsPackable>false</IsPackable>
+    <LangVersion>Latest</LangVersion>
+    <ManagePackageVersionsCentrally>True</ManagePackageVersionsCentrally>
+    <NoWarn>CS1591</NoWarn>
+    <Nullable>enable</Nullable>
+    <ProjectRoot>$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), "*.sln"))</ProjectRoot>
+    <TargetFramework>net7.0</TargetFramework>
+    <TreatWarningsAsErrors>False</TreatWarningsAsErrors>
+  </PropertyGroup>
+  <!-- Nuget -->
```
Comment (akiraveliara) [inline Directory.Build.props:17]: there should be a newline

File: Directory.Build.props [hunk 2]
```diff
@@ -1,73 +1,36 @@
...
-    <PropertyGroup>
-      <_RemoveAnalyzer>%(RemoveAnalyzer.Identity)</_RemoveAnalyzer>
-    </PropertyGroup>
-    <ItemGroup>
-      <Analyzer Remove="$(_RemoveAnalyzer)"/>
-      <Analyzer Remove="@(Analyzer)" Condition="'%(Analyzer.Filename)' == '$(_RemoveAnalyzer)'"/>
-    </ItemGroup>
-  </Target>
-</Project>
+  <ItemGroup>
```
Comment (akiraveliara) [inline Directory.Build.props:36]: there should be a newline here too

File: DSharpPlus.VoiceNext/DSharpPlus.VoiceNext.csproj [hunk 3]
```diff
@@ -1,23 +1,13 @@
...
+  <PropertyGroup>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <PackageId>DSharpPlus.VoiceNext</PackageId>
+    <Description>Voice implementation for DSharpPlus.</Description>
+    <PackageTags>$(PackageTags), audio, voice, radio, music</PackageTags>
+    <IsPackable>true</IsPackable>
+  </PropertyGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\DSharpPlus\DSharpPlus.csproj"/>
+    <PackageReference Include="DSharpPlus.VoiceNext.Natives"/>
```
Comment (akiraveliara) [inline DSharpPlus.VoiceNext/DSharpPlus.VoiceNext.csproj:11]: having both project and package references in one item group defies convention; even `dotnet add` should separate them

File: DSharpPlus.SlashCommands/DSharpPlus.SlashCommands.csproj [hunk 4]
```diff
@@ -1,23 +1,12 @@
...
-</Project>
+  <PropertyGroup>
+    <PackageId>DSharpPlus.SlashCommands</PackageId>
+    <Description>An extension for DSharpPlus to make slash commands easy</Description>
+    <PackageTags>$(PackageTags), commands, slash-commands, interactions</PackageTags>
+    <Authors>IDoEverything, Epictek, SakuraIsayeki, sssvt-drabek-stepan, tygore587, VelvetThePanda, OoLunar</Authors>
+  </PropertyGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\DSharpPlus\DSharpPlus.csproj"/>
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection"/>
```
Comment (akiraveliara) [inline DSharpPlus.SlashCommands/DSharpPlus.SlashCommands.csproj:10]: same hdre

File: DSharpPlus.Lavalink/DSharpPlus.Lavalink.csproj [hunk 5]
```diff
@@ -1,19 +1,11 @@
...
-    <ItemGroup>
-        <ProjectReference Include="..\DSharpPlus\DSharpPlus.csproj"/>
-    </ItemGroup>
-
-</Project>
+  <PropertyGroup>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <PackageId>DSharpPlus.Lavalink</PackageId>
+    <Description>Lavalink implementation for DSharpPlus. Lavalink is an easy and lightweight method of streaming audio to Discord.</Description>
+    <PackageTags>discord, discord-api, bots, discord-bots, chat, dsharp, dsharpplus, csharp, dotnet, vb-net, fsharp, audio, voice, radio, music, lavalink, lavaplayer</PackageTags>
```
Comment (akiraveliara) [inline DSharpPlus.Lavalink/DSharpPlus.Lavalink.csproj:6]: why doesn't this use the global tags?

File: DSharpPlus.Interactivity/DSharpPlus.Interactivity.csproj [hunk 6]
```diff
@@ -1,22 +1,12 @@
...
+    <IsPackable>true</IsPackable>
   </PropertyGroup>
-
-  <ItemGroup>
-    <PackageReference Include="ConcurrentHashSet"/>
-  </ItemGroup>
-
   <ItemGroup>
     <ProjectReference Include="..\DSharpPlus\DSharpPlus.csproj" />
+    <PackageReference Include="ConcurrentHashSet"/>
```
Comment (akiraveliara) [inline DSharpPlus.Interactivity/DSharpPlus.Interactivity.csproj:10]: convention again here

File: DSharpPlus.CommandsNext/DSharpPlus.CommandsNext.csproj [hunk 7]
```diff
@@ -1,24 +1,13 @@
...
   <PropertyGroup>
     <PackageId>DSharpPlus.CommandsNext</PackageId>
     <Description>Advanced command framework for DSharpPlus.</Description>
-    <PackageTags>discord, discord-api, bots, discord-bots, chat, dsharp, dsharpplus, csharp, dotnet, vb-net, fsharp, commands, commandsnext</PackageTags>
+    <PackageTags>$(PackageTags), commands, commandsnext</PackageTags>
+    <IsPackable>true</IsPackable>
   </PropertyGroup>
-
   <ItemGroup>
+    <ProjectReference Include="..\DSharpPlus\DSharpPlus.csproj" />
```
Comment (akiraveliara) [inline DSharpPlus.CommandsNext/DSharpPlus.CommandsNext.csproj:9]: and once more

File: Directory.Packages.props [hunk 8]
```diff
@@ -6,7 +6,7 @@
     <PackageVersion Include="Microsoft.Extensions.Caching.Memory" Version="7.0.0"/>
     <PackageVersion Include="Microsoft.Extensions.DependencyInjection" Version="7.0.0"/>
     <PackageVersion Include="Microsoft.Extensions.DependencyModel" Version="7.0.0"/>
-    <PackageVersion Include="Microsoft.Extensions.Logging.Abstractions" Version="7.0.0"/>
+    <PackageVersion Include="Microsoft.Extensions.Logging.Abstractions" Version="7.0.1"/>
```
Comment (akiraveliara) [inline Directory.Packages.props:9]: generally please dont update dependencies in unrelated PRs

**Review by akiraveliara — CHANGES_REQUESTED** [review]
why are IsPackable and OutputType not in Directory.Build.props?

File: Directory.Packages.props (hunk 8 above)
Comment (OoLunar) [reply Directory.Packages.props:9]: I only did it because it was a minor patch

File: Directory.Build.props (hunk 1 above)
Comment (OoLunar) [reply Directory.Build.props:18]: My formatter literally does not allow for this. I cannot save the file without the formatter removing the newlines.

File: DSharpPlus.VoiceNext/DSharpPlus.VoiceNext.csproj (hunk 3 above)
Comment (OoLunar) [reply DSharpPlus.VoiceNext/DSharpPlus.VoiceNext.csproj:11]: `dotnet add` doesn't even support project references iirc... I also fail to see why it's convention in the first place.

File: DSharpPlus.Lavalink/DSharpPlus.Lavalink.csproj (hunk 5 above)
Comment (OoLunar) [reply DSharpPlus.Lavalink/DSharpPlus.Lavalink.csproj:6]: Oops missed a spot

**Comment by OoLunar** [comment]
> why are IsPackable and OutputType not in Directory.Build.props?

It is? [OutputType](https://github.com/DSharpPlus/DSharpPlus/blob/706613538b297899b57821242c717f92c238f87e/Directory.Build.props#L10), [IsPackable](https://github.com/DSharpPlus/DSharpPlus/blob/706613538b297899b57821242c717f92c238f87e/Directory.Build.props#L5)

File: Directory.Build.props (hunk 1 above)
Comment (akiraveliara) [reply Directory.Build.props:18]: that is an issue unto its own, but does not excuse not doing it. if theres a separate reason then maybe

**Comment by akiraveliara** [comment]
i meant IsPackable=true, but with OutputType there you might want to give the csprojs another pass if you haven't already

File: DSharpPlus.VoiceNext/DSharpPlus.VoiceNext.csproj (hunk 3 above)
Comment (akiraveliara) [reply DSharpPlus.VoiceNext/DSharpPlus.VoiceNext.csproj:11]: `dotnet add reference`, and while i don't know, it is what all tooling in the world uses, so it's probably better to keep that way

File: Directory.Build.props (hunk 1 above)
Comment (OoLunar) [reply Directory.Build.props:18]: `nano` it is then...

File: DSharpPlus.VoiceNext/DSharpPlus.VoiceNext.csproj (hunk 3 above)
Comment (OoLunar) [reply DSharpPlus.VoiceNext/DSharpPlus.VoiceNext.csproj:11]: All the tooling probably also uses `dotnet add reference` then. And seeing as how there is no functional difference at the moment, are you *sure*  you want me to revert that change?

Comment (akiraveliara) [reply DSharpPlus.VoiceNext/DSharpPlus.VoiceNext.csproj:13]: VS doesn't, or at least didn't; unsure about rider. if dotnet add just works with this then *maybe,* if dotnet add creates new groups then absolutely revert it

**Review by akiraveliara — APPROVED** [review]

**Pull request merged**